docs.go

```diff
@@ -7,13 +7,94 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"os"
+	"regexp"
 	"sort"
+	"strconv"
 	"strings"
 	"text/template"
+	"unicode/utf8"
 
 	"github.com/cpuguy83/go-md2man/v2/md2man"
 )
 
+// ToTabularMarkdown creates a tabular markdown documentation for the `*App`.
+// The function errors if either parsing or writing of the string fails.
+func (a *App) ToTabularMarkdown(appPath string) (string, error) {
+	if appPath == "" {
+		appPath = "app"
+	}
+
+	const name = "cli"
+
+	t, err := template.New(name).Funcs(template.FuncMap{
+		"join": strings.Join,
+	}).Parse(MarkdownTabularDocTemplate)
+	if err != nil {
+		return "", err
+	}
+
+	var (
+		w  bytes.Buffer
+		tt tabularTemplate
+	)
+
+	if err = t.ExecuteTemplate(&w, name, cliTabularAppTemplate{
+		AppPath:     appPath,
+		Name:        a.Name,
+		Description: tt.PrepareMultilineString(a.Description),
+		Usage:       tt.PrepareMultilineString(a.Usage),
+		UsageText:   strings.FieldsFunc(a.UsageText, func(r rune) bool { return r == '\n' }),
+		ArgsUsage:   tt.PrepareMultilineString(a.ArgsUsage),
+		GlobalFlags: tt.PrepareFlags(a.VisibleFlags()),
+		Commands:    tt.PrepareCommands(a.VisibleCommands(), appPath, "", 0),
+	}); err != nil {
+		return "", err
```

Comment:
can you add a test case where the template execution returns an error ?

Reply:
Done in d945e8d754cbecdc07c4a496a6911b0f41522f63